src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/json/EventTransformer.scala

```diff
@@ -67,23 +67,95 @@ object EventTransformer {
    * @return ValidatedRecord containing JSON for the event and the event_id (if it exists)
    */
   def jsonifyGoodEvent(event: Array[String]): Validated[(Set[InventoryItem], JObject)] = {
-    if (event.length != Fields.size) {
+    getValidatedJsonEvent(event)
+  }
+
+  /**
+    * Converts an array of field values to a validated JSON whose keys are the field names corresponding to the
+    * EnrichedEvent POJO of the Scala Common Enrich project. If there are any self-describing events in the fields:
+    * "unstruct_event", "derived_contexts" or "contexts" these are returned in a "shredded" format (e.g.
+    * "unstruct_event_com_acme_1_myField": "value"). This default behaviour can also be disabled, so that the deep
+    * structure is returned for those three fields instead, by setting the flatten parameter to false.
+    *
+    * @param event Array of values for the event
+    * @param flatten Whether to flattent the fields in "unstruct_event", "contexts" and "derived_contexts"
+    * @return ValidatedRecord containing JSON for the event and the event_id (if it exists)
+    */
+  def getValidatedJsonEvent(event: Array[String], flatten:Boolean = true): Validated[(Set[InventoryItem], JObject)] = {
```

Comment:
Adding default parameters is binary incompatible change (but source-compatible, which makes it fail in runtime with `NoSuchMethodError`): https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html#evolving-code-without-breaking-binary-compatibility. This should be defined as separate function.

Reply:
Good catch, anyway boolean args are not great. 👍

Reply:
> anyway boolean args are not great

Agreed. I'm planning to either make format an ADT or implement as separate functions in 0.4.0. What option is more preferable to you?